src/app/public/modules/back-to-top/back-to-top.directive.ts

```diff
@@ -69,21 +90,32 @@ export class SkyBackToTopDirective implements AfterViewInit, OnDestroy {
       });
   }
 
-  private setBackToTopListener(): void {
+  private handleIncomingMessages(message: SkyBackToTopMessage): void {
+    /* istanbul ignore else */
+    if (message.type === SkyBackToTopMessageType.BackToTop) {
+      this.domAdapter.scrollToElement(this.element);
+    }
+  }
+
+  private setBackToTopListeners(): void {
     /* istanbul ignore else */
     if (this.element) {
       this.domAdapter.elementInViewOnScroll(this.element)
         .pipe(takeUntil(this.ngUnsubscribe))
         .subscribe((elementInView: boolean) => {
           // Add back to top button if user scrolls down.
-          if (!this.dockItem && !elementInView) {
+          if (!this.dockItem && !elementInView && !this.hideBackToTopButton) {
             this.addBackToTop();
           }
           // Remove back to top button if user scrolls back up.
           if (elementInView) {
             this.destroyBackToTop();
           }
       });
+
+      this.skyBackToTopMessageStream
```

Comment:
Since this only happens in `ngAfterViewInit()`, it violates our "once-ability" principle where subsequent updates to the property won't be respected. Adding the subscription should be done in a `set`ter for the `skyBackToTopMessageStream` property, and if it's already set, you should unsubscribe before reassigning the local field and subscribing.